Dockerfile

```diff
@@ -47,4 +47,7 @@ RUN git clone --depth 1 https://github.com/pyenv/pyenv.git $PYENV_HOME \
     && pyenv install 3.9.4 \
     && pyenv global system 3.6.13 3.7.10 3.8.9 3.9.4
 
+# Install the latest Poetry
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
```

Comment:
Personally, I think it's bad security practice to `curl <script> | execute` random scripts from the internet. What's the benefit of this over `pip install poetry` from pypi?

Reply:
I 100% agree. You are absolutely correct there. Unfortunately, that is the official method for installing poetry. https://python-poetry.org/docs/#installation

According to the docs:
> Using alternative installation methods will make Poetry always use the Python version for which it has been installed to create virtualenvs.
> 
> So, you will need to install Poetry for each Python version you want to use and switch between them.

Reply:
Thanks for the explanation. I guess this is fine if that's the recommended way. I suppose a way we can secure this is to record the checksum and compare it to make sure we are getting the version of the script we expect but I'm not too sure how feasible to maintain if that script changes often. I'm going to give my approval here with that in mind.

Reply:
I was actually thinking along those same lines as well. In fact there is an issue already filed against Poetry to set up basically just that: https://github.com/python-poetry/poetry/issues/3442